File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/admin/service/UserService.java
package org.bootstrapbugz.api.admin.service;

import org.bootstrapbugz.api.admin.payload.request.PatchUserRequest;
import org.bootstrapbugz.api.admin.payload.request.UserRequest;
import org.bootstrapbugz.api.shared.generic.crud.CrudService;
import org.bootstrapbugz.api.user.payload.dto.UserDTO;

public interface UserService extends CrudService<UserDTO, UserRequest> {
  UserDTO patch(Long id, PatchUserRequest patchUserRequest);
}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/admin/service/UserService.kt
package org.bootstrapbugz.api.admin.service

import org.bootstrapbugz.api.admin.payload.request.PatchUserRequest
import org.bootstrapbugz.api.admin.payload.request.UserRequest
import org.bootstrapbugz.api.shared.generic.crud.CrudService
import org.bootstrapbugz.api.user.payload.dto.UserDTO

interface UserService : CrudService<UserDTO, UserRequest> {
    fun patch(id: Long?, patchUserRequest: PatchUserRequest): UserDTO
}
